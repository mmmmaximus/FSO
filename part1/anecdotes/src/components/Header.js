import React from 'react'

const Header = ({text}) => <h1 data-testid='Header'>{text}</h1>

export default Header
