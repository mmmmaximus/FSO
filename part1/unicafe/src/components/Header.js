import React from 'react'

const Header = ({header}) => <h1 data-testid='Header'>{header}</h1>

export default Header
