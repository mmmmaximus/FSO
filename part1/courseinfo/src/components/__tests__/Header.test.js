import { render, screen } from '@testing-library/react'
import Header from '../Header'
import renderer from 'react-test-renderer'

it('should render Header component', () => {
  const course = {
    name: 'Half Stack application development',
    parts: [
      {
        name: 'Fundamentals of React',
        exercises: 10
      },
      {
        name: 'Using props to pass data',
        exercises: 7
      },
      {
        name: 'State of a component',
        exercises: 14
      }
    ]
  }
  render(<Header course={course.name}/>)
  const HeaderElement = screen.getByTestId('Header')
  expect(HeaderElement).toBeInTheDocument
  expect(HeaderElement.textContent).toBe('Half Stack application development')
})

test('matches snapshot', () => {
  const course = {
    name: 'Half Stack application development',
    parts: [
      {
        name: 'Fundamentals of React',
        exercises: 10
      },
      {
        name: 'Using props to pass data',
        exercises: 7
      },
      {
        name: 'State of a component',
        exercises: 14
      }
    ]
  }
  const tree = renderer.create(<Header course={course.name}/>).toJSON()
  expect(tree).toMatchSnapshot()
})
