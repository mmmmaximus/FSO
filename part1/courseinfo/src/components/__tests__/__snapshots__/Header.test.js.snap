// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<h1
  data-testid="Header"
>
  Half Stack application development
</h1>
`;
